oci-deployment-examples/oci-deployment-shell-stages/README.md

```diff
@@ -0,0 +1,346 @@
+Sample illustration of OCI DevOps deployment pipeline with *DevOps Deploy Shell stage*
+------------
+
+## Deploy Shell Stage
+The shell stage allows you to run custom commands in the deployment pipeline. This stage can be added at any point in the deployment pipeline. You must prepare a command specification YAML file containing all the steps that you want to run during the deployment. The command spec is run on a container instance host in the selected compartment and subnet in your tenancy.
+
+⚠️ - The feature is currently in `limited availability` mode.Please contact your Oracle representative to get it enabled for your tenancy.
+
+## Key benefits of adding a Shell stage:
+
+- Automate deployments to OCI platforms such as service mesh and container instances using OCI CLI and other pre-installed tools.
+
+- OCI CLI is pre-authenticated to use the pipeline resource principal to access OCI resources.
+
+- Run a database schema migration using command line tools such as Oracle SQL Developer Command Line (SQLcl).
+
+- Use any pipeline parameters by referencing them in the spec. Also, replace parameters and variables during the deployment run.
+
+- Perform integration tests on artifacts before deploying to production.
+
+- Customize the DevOps workflow. For example, you can wait for some resources to be ready before the pipeline deploys additional resources.
+
+Specific instructions to download only this sample.
+---
+
+```
+    $ git init oci-deployment-shell-stages
+    $ cd oci-deployment-shell-stages
+    $ git remote add origin https://github.com/oracle-devrel/oci-devops-examples
+    $ git config core. sparsecheckout true
+    $ echo "oci-deployment-examples/ooci-deployment-shell-stages/*">>.git/info/sparse-checkout
+    $ git pull --depth=1 origin main
+```
+
+
+## Sample 1 . Simple Shell stage execution.
+
+### Network requirement.
+
+- A valid OCI Virtual cloud network (OCI VCN) with at least one subnet.
+- The subnet will be used by the container instances to run the shell stage executions.
+
+### OCI Identity setup
+
+- Create a dynamic group with the below rule.
+
+```java
+ALL {resource.type = 'devopsdeploypipeline', resource.compartment.id = 'COMPARTMENT_OCID'}
+```
+
+- Create a dynamic policy with the below statements.
+
+```java
+Allow dynamic-group  <NAME OF THE DYNAMIC GROUP> to manage virtual-network-family in compartment <NAME OF THE COMPARTMENT> 
+Allow dynamic-group <NAME OF THE DYNAMIC GROUP>  to manage ons-topics in compartment <NAME OF THE COMPARTMENT>
+Allow dynamic-group <NAME OF THE DYNAMIC GROUP>  to manage all-artifacts in compartment <NAME OF THE COMPARTMENT> 
```

Comment:
We need to have policy for compute-containers as well. Either "to manage compute-container-family" or specific policies as mentioned in TC doc.
https://confluence.oci.oraclecorp.com/display/DLCDEP/TC+documentation+details+-+Managed+Shell

Reply:
thank you for the review ,
the policy  statement is added